taichi/ir/snode.cpp

```diff
@@ -8,6 +8,18 @@ TLANG_NAMESPACE_BEGIN
 
 std::atomic<int> SNode::counter = 0;
 
+SNode &SNode::insert_children(SNodeType t) {
+  TI_ASSERT(t != SNodeType::root);
+
+  auto new_ch = std::make_unique<SNode>(depth + 1, t);
+  new_ch->is_path_all_dense = (is_path_all_dense && ((t == SNodeType::dense) ||
+                                                     (t == SNodeType::place)));
+  ch.push_back(std::move(new_ch));
+  // Note: |new_ch->parent| will not be set until structural nodes are compiled!
+  // (But why..?)
```

Comment:
I think it's because in the past we change the SNode hierarchy during the SNode compilation phase, so `snode->parent` will change there. I'm not sure if that's still the case now...

Reply:
Ah ok, let's worry about this later..